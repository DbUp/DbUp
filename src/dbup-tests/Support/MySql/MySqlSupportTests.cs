#if !NETCORE
using System;
using DbUp.Tests.TestInfrastructure;
using Shouldly;
using Xunit;

namespace DbUp.Tests.Support.MySql
{
    public class MySqlSupportTests
    {
        [Fact]
        public void CanHandleDelimiter()
        {
            var logger = new CaptureLogsLogger();
            var recordingDbConnection = new RecordingDbConnection(logger, "schemaversions");
            recordingDbConnection.SetupRunScripts();
            var upgrader = DeployChanges.To
                .MySqlDatabase(string.Empty)
                .OverrideConnectionFactory(recordingDbConnection)
                .LogTo(logger)
                .WithScript("Script0003", @"USE `test`;
DROP procedure IF EXISTS `testSproc`;

DELIMITER $$

USE `test`$$
CREATE PROCEDURE `testSproc`(
        IN   ssn                    VARCHAR(32)
     )
BEGIN 

    SELECT id      
    FROM   customer as c
    WHERE  c.ssn = ssn ; 

END$$").Build();

            var result = upgrader.PerformUpgrade();

            result.Successful.ShouldBe(true);
            logger.Log.ShouldMatchApproved(b => b.WithScrubber(Scrubbers.ScrubDates));
        }
    }
}
#endif