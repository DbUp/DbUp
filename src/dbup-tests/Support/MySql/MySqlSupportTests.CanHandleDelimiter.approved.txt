DB Operation: Open connection
Info:         Beginning database upgrade
Info:         Checking whether journal table exists..
DB Operation: Execute scalar command: select 1 from INFORMATION_SCHEMA.TABLES where TABLE_NAME = 'schemaversions'
DB Operation: Dispose command
Info:         Fetching list of already executed scripts.
DB Operation: Execute reader command: select scriptname from `schemaversions` order by scriptname
DB Operation: Dispose command
Info:         Executing Database Server script 'Script0003'
Info:         Checking whether journal table exists..
DB Operation: Execute scalar command: select 1 from INFORMATION_SCHEMA.TABLES where TABLE_NAME = 'schemaversions'
DB Operation: Dispose command
DB Operation: Execute non query command: USE `test`
DB Operation: Dispose command
DB Operation: Execute non query command: DROP procedure IF EXISTS `testSproc`
DB Operation: Dispose command
DB Operation: Execute non query command: USE `test`
DB Operation: Dispose command
DB Operation: Execute non query command: CREATE PROCEDURE `testSproc`(
        IN   ssn                    VARCHAR(32)
     )
BEGIN 

    SELECT id      
    FROM   customer as c
    WHERE  c.ssn = ssn ; 

END
DB Operation: Dispose command
DB Operation: Create parameter
Info:         DB Operation: Add parameter to command: scriptName=Script0003
DB Operation: Create parameter
Info:         DB Operation: Add parameter to command: applied=<date>
DB Operation: Create parameter
Info:         DB Operation: Add parameter to command: hash=
DB Operation: Execute non query command: insert into `schemaversions` (ScriptName, Applied) values (@scriptName, @applied)
DB Operation: Dispose command
Info:         Upgrade successful
DB Operation: Dispose connection
